code/game/objects/items/weapons/tools.dm

```diff
@@ -791,26 +791,52 @@
 		else
 			to_chat(user, SPAN_WARNING("The cell isn't charged!"))
 
+/obj/item/steelwool/fire_act()
+	ignite()
+
+/obj/item/steelwool/afterattack(atom/A, mob/user)
+	if(istype(A, /obj) || istype(A, /turf))
+		file_down(A, user)
+	return
 
 /obj/item/steelwool/proc/ignite(var/L, mob/user)
-	lit = TRUE
-	user.visible_message(SPAN_NOTICE("[user] ignites the steel wool with \the [L]."), SPAN_NOTICE("You ignite the steel wool."), SPAN_NOTICE("You hear a gentle flame crackling."))
-	playsound(get_turf(src), 'sound/items/flare.ogg', 50)
-	desc += " Watch your hands!"
-	icon_state = "burning_wool"
-	set_light(2, 2, LIGHT_COLOR_LAVA)
-	addtimer(CALLBACK(src, .proc/endburn, user), 120 SECONDS, TIMER_UNIQUE)
-
-/obj/item/steelwool/proc/endburn(mob/living/carbon/human/user)
+	if(lit && user)
+		to_chat(user, SPAN_WARNING("The steel wool is already lit!"))
+		return
+	else
+		lit = TRUE
+		if(user)
+			user.visible_message(SPAN_NOTICE("[user] ignites the steel wool with \the [L]."), SPAN_NOTICE("You ignite the steel wool with \the [L]"), SPAN_NOTICE("You hear a gentle flame crackling."))
+		playsound(get_turf(src), 'sound/items/flare.ogg', 50)
+		desc += " Watch your hands!"
+		icon_state = "burning_wool"
+		set_light(2, 2, LIGHT_COLOR_LAVA)
+		addtimer(CALLBACK(src, .proc/endburn), 120 SECONDS, TIMER_UNIQUE)
+
+/obj/item/steelwool/proc/endburn()
 	visible_message(SPAN_NOTICE("The steel wool burns out."))
-	if(!user.gloves)
-		var/UserLoc = get_equip_slot()
-		if(UserLoc == slot_l_hand)
-			user.apply_damage(5, BURN, BP_L_HAND)
-			to_chat(user, SPAN_DANGER("The steel wool burns your left hand!"))
-		else if(UserLoc == slot_r_hand)
-			user.apply_damage(5, BURN, BP_R_HAND)
-			to_chat(user, SPAN_DANGER("The steel wool burns your right hand!"))
+	if(istype(loc, /mob/living))
+		var/mob/living/carbon/human/user = loc
+		if(!user.gloves)
+			var/UserLoc = get_equip_slot()
+			if(UserLoc == slot_l_hand)
+				user.apply_damage(5, BURN, BP_L_HAND)
+				to_chat(user, SPAN_DANGER("The steel wool burns your left hand!"))
+			else if(UserLoc == slot_r_hand)
+				user.apply_damage(5, BURN, BP_R_HAND)
+				to_chat(user, SPAN_DANGER("The steel wool burns your right hand!"))
```

Comment:
Don't understand why we're burning hands when the steel wool stops burning, rather than whenever someone picks it up / is holding it?

Why isn't it burning them in the 120 seconds since the steel wool was ignited?